src/detect-gid.c

```diff
@@ -73,8 +73,9 @@ static int DetectGidSetup (DetectEngineCtx *de_ctx, Signature *s, const char *ra
 {
     unsigned long gid = 0;
     char *endptr = NULL;
+    errno = 0;
```

Comment:
Do we really need to set this before ?

Reply:
yes, see https://github.com/OISF/suricata/pull/11456#discussion_r1670506278